contract/vm.go

```diff
@@ -239,7 +239,7 @@ func NewVmContextQuery(
 }
 
 func (ctx *vmContext) IsGasSystem() bool {
-	return !ctx.isQuery && PubNet && ctx.blockInfo.ForkVersion >= 2
+	return fee.IsVmGasSystem(ctx.blockInfo.ForkVersion, ctx.isQuery)
```

Comment:
`fee.IsVmGasSystem` is not using `PubNet`

Reply:
pubnet is just reverse value of zerofee. i think check zerofee is more appropriate than pubnet.

Reply:
Ah, it is replaced by `IsZeroFee`